commit bfa363472bfb861a02bdeefc7477fcab04091c66
Author: Anthony Sottile <asottile@umich.edu>
Date:   Mon Aug 19 15:50:31 2024 -0400

    avoid interleaving pywsgi threadstate

diff --git a/plugins/pyuwsgi/pyuwsgi.c b/plugins/pyuwsgi/pyuwsgi.c
index 7a4f2249..11732e04 100644
--- a/plugins/pyuwsgi/pyuwsgi.c
+++ b/plugins/pyuwsgi/pyuwsgi.c
@@ -126,13 +126,6 @@ PyObject *pyuwsgi_setup(PyObject * self, PyObject * args, PyObject * kwds) {
 		return NULL;
 	}
 
-
-	//TODO: ...???
-	// actually do the thing!
-	PyThreadState *_tstate = PyThreadState_Get();
-	uwsgi_setup(orig_argc, orig_argv, environ);
-	PyThreadState_Swap(_tstate);
-
 	Py_INCREF(self);
 	return self;
 }
@@ -143,6 +136,7 @@ PyObject *pyuwsgi_init(PyObject * self, PyObject * args, PyObject * kwds) {
 		return NULL;
 	}
 
+	uwsgi_setup(orig_argc, orig_argv, environ);
 	int rc = uwsgi_run();
 
 	// never(?) here
@@ -156,6 +150,7 @@ PyObject *pyuwsgi_run(PyObject * self, PyObject * args, PyObject * kwds) {
 		return NULL;
 	}
 
+	uwsgi_setup(orig_argc, orig_argv, environ);
 	int rc = uwsgi_run();
 
 	// never(?) here
